### PR TITLE
Double registers per iteration in FP32-arithmetic FP16 ARM gemv kernel

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127033
* #127016
* __->__ #126877
* #126794

Summary: I found that doubling this significantly improved performance, but doubling again did not, so I stopped here.

Test Plan: CI
Benchmarked with llm_experiments repo as previously in stack; relevant data:

before:
trans_b torch.float16 1396.11 usec (4100)
trans_b torch.float16 1399.54 usec (4104)

after:
trans_b  torch.float16 1096.00 usec (4100)
trans_b  torch.float16 1093.47 usec (4104)

### CODE REVIEW

**Comment by swolchok** [comment]
@swolchok has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D57732735).

**Review by malfet — APPROVED** [review]

File: aten/src/ATen/native/BlasKernel.cpp [hunk 1]
```diff
@@ -355,6 +355,10 @@ static inline double reduce(float32x4_t x[F32_REGISTERS_PER_ITERATION]) {
   for (int i = 0; i < offset; ++i) {
     x[i] = vaddq_f32(x[i], x[offset + i]);
   }
+  offset /= 2;
```
[the remark this answers is not in the document]
Comment (swolchok) [reply aten/src/ATen/native/BlasKernel.cpp:358]: https://github.com/pytorch/pytorch/pull/127016

**Pull request closed without merging**